### ISSUE TITLE
nextcloud update: General error: 1 no such table: oc_appconfig

### ISSUE THREAD

**Issue by teto** [NixOS/nixpkgs#67323]
**Describe the bug**
I have been trying to upgrade my nextcloud server via nixops but activation fails because of a nextcloud error:

systemctl status nextcloud-setup.service                                                                                                                                                                                                                         
```
● nextcloud-setup.service
   Loaded: loaded (/nix/store/ny7pszv7x741kmqamsxhhiz2haxizlmc-unit-nextcloud-setup.service/nextcloud-setup.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2019-08-23 14:38:08 JST; 25s ago
  Process: 2208 ExecStart=/nix/store/vadyi9z59gybajxdgv9xb2mxcslawpws-unit-script-nextcloud-setup-start (code=exited, status=1/FAILURE)
 Main PID: 2208 (code=exited, status=1/FAILURE)

Aug 23 14:38:08 gitolite vadyi9z59gybajxdgv9xb2mxcslawpws-unit-script-nextcloud-setup-start[2208]: #25 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(114): OC_App::getEnabledApps()
Aug 23 14:38:08 gitolite vadyi9z59gybajxdgv9xb2mxcslawpws-unit-script-nextcloud-setup-start[2208]: #26 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/base.php(640): OC_App::loadApps(Array)
Aug 23 14:38:08 gitolite vadyi9z59gybajxdgv9xb2mxcslawpws-unit-script-nextcloud-setup-start[2208]: #27 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/base.php(1056): OC::init()
Aug 23 14:38:08 gitolite vadyi9z59gybajxdgv9xb2mxcslawpws-unit-script-nextcloud-setup-start[2208]: #28 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/console.php(46): require_once('/nix/store/i2xy...')
Aug 23 14:38:08 gitolite vadyi9z59gybajxdgv9xb2mxcslawpws-unit-script-nextcloud-setup-start[2208]: #29 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/occ(11): require_once('/nix/store/i2xy...')
Aug 23 14:38:08 gitolite sudo[2214]: pam_unix(sudo:session): session closed for user nextcloud
Aug 23 14:38:08 gitolite vadyi9z59gybajxdgv9xb2mxcslawpws-unit-script-nextcloud-setup-start[2208]: #30 {main}
Aug 23 14:38:08 gitolite systemd[1]: nextcloud-setup.service: Main process exited, code=exited, status=1/FAILURE
Aug 23 14:38:08 gitolite systemd[1]: nextcloud-setup.service: Failed with result 'exit-code'.
Aug 23 14:38:08 gitolite systemd[1]: Failed to start nextcloud-setup.service.

```

I tried to run manually the script launched by the systemd service and here is the error
```
/nix/store/vadyi9z59gybajxdgv9xb2mxcslawpws-unit-script-nextcloud-setup-start                                                                                                                                                                                    
An unhandled exception has been thrown:
PDOException: SQLSTATE[HY000]: General error: 1 no such table: oc_appconfig in /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:107
Stack trace:
#0 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php(107): PDO->query('SELECT * FROM "...')
#1 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(935): Doctrine\DBAL\Driver\PDOConnection->query('SELECT * FROM "...')
#2 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/DB/Connection.php(195): Doctrine\DBAL\Connection->executeQuery('SELECT * FROM "...', Array, Array, NULL)
#3 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php(219): OC\DB\Connection->executeQuery('SELECT * FROM "...', Array, Array)
#4 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/DB/QueryBuilder/QueryBuilder.php(214): Doctrine\DBAL\Query\QueryBuilder->execute()
#5 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppConfig.php(316): OC\DB\QueryBuilder\QueryBuilder->execute()
#6 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppConfig.php(94): OC\AppConfig->loadConfigValues()
#7 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppConfig.php(272): OC\AppConfig->getApps()
#8 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(882): OC\AppConfig->getValues(false, 'installed_versi...')
#9 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(501): OC_App::getAppVersions()
#10 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(118): OC\Server->OC\{closure}(Object(OC\Server))
#11 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('OC\\Memcache\\Fac...')
#12 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppFramework/Utility/SimpleContainer.php(165): OC\ServerContainer->query('OC\\Memcache\\Fac...')
#13 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(114): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#14 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('MemCacheFactory')
#15 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(1541): OC\ServerContainer->query('MemCacheFactory')
#16 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(697): OC\Server->getMemCacheFactory()
#17 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(118): OC\Server->OC\{closure}(Object(OC\Server))
#18 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('OC\\App\\AppManag...')
#19 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppFramework/Utility/SimpleContainer.php(165): OC\ServerContainer->query('OC\\App\\AppManag...')
#20 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(114): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#21 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('AppManager')
#22 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(1731): OC\ServerContainer->query('AppManager')
#23 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(343): OC\Server->getAppManager()
#24 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(114): OC_App::getEnabledApps()
#25 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/base.php(640): OC_App::loadApps(Array)
#26 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/base.php(1056): OC::init()
#27 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/console.php(46): require_once('/nix/store/i2xy...')
#28 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/occ(11): require_once('/nix/store/i2xy...')
#29 {main}

Next Doctrine\DBAL\Driver\PDOException: SQLSTATE[HY000]: General error: 1 no such table: oc_appconfig in /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:109
Stack trace:
#0 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(935): Doctrine\DBAL\Driver\PDOConnection->query('SELECT * FROM "...')
#1 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/DB/Connection.php(195): Doctrine\DBAL\Connection->executeQuery('SELECT * FROM "...', Array, Array, NULL)
#2 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php(219): OC\DB\Connection->executeQuery('SELECT * FROM "...', Array, Array)
#3 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/DB/QueryBuilder/QueryBuilder.php(214): Doctrine\DBAL\Query\QueryBuilder->execute()
#4 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppConfig.php(316): OC\DB\QueryBuilder\QueryBuilder->execute()
#5 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppConfig.php(94): OC\AppConfig->loadConfigValues()
#6 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppConfig.php(272): OC\AppConfig->getApps()
#7 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(882): OC\AppConfig->getValues(false, 'installed_versi...')
#8 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(501): OC_App::getAppVersions()
#9 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(118): OC\Server->OC\{closure}(Object(OC\Server))
#10 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('OC\\Memcache\\Fac...')
#11 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppFramework/Utility/SimpleContainer.php(165): OC\ServerContainer->query('OC\\Memcache\\Fac...')
#12 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(114): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#13 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('MemCacheFactory')
#14 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(1541): OC\ServerContainer->query('MemCacheFactory')
#15 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(697): OC\Server->getMemCacheFactory()
#16 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(118): OC\Server->OC\{closure}(Object(OC\Server))
#17 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('OC\\App\\AppManag...')
#18 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppFramework/Utility/SimpleContainer.php(165): OC\ServerContainer->query('OC\\App\\AppManag...')
#19 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(114): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#20 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('AppManager')
#21 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(1731): OC\ServerContainer->query('AppManager')
#22 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(343): OC\Server->getAppManager()
#23 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(114): OC_App::getEnabledApps()
#24 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/base.php(640): OC_App::loadApps(Array)
#25 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/base.php(1056): OC::init()
#26 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/console.php(46): require_once('/nix/store/i2xy...')
#27 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/occ(11): require_once('/nix/store/i2xy...')
#28 {main}

Next Doctrine\DBAL\Exception\TableNotFoundException: An exception occurred while executing 'SELECT * FROM "oc_appconfig"':

SQLSTATE[HY000]: General error: 1 no such table: oc_appconfig in /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractSQLiteDriver.php:63
Stack trace:
#0 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php(184): Doctrine\DBAL\Driver\AbstractSQLiteDriver->convertException('An exception oc...', Object(Doctrine\DBAL\Driver\PDOException))
#1 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php(158): Doctrine\DBAL\DBALException::wrapException(Object(Doctrine\DBAL\Driver\PDOSqlite\Driver), Object(Doctrine\DBAL\Driver\PDOException), 'An exception oc...')
#2 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(938): Doctrine\DBAL\DBALException::driverExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDOSqlite\Driver), Object(Doctrine\DBAL\Driver\PDOException), 'SELECT * FROM "...', Array)
#3 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/DB/Connection.php(195): Doctrine\DBAL\Connection->executeQuery('SELECT * FROM "...', Array, Array, NULL)
#4 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php(219): OC\DB\Connection->executeQuery('SELECT * FROM "...', Array, Array)
#5 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/DB/QueryBuilder/QueryBuilder.php(214): Doctrine\DBAL\Query\QueryBuilder->execute()
#6 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppConfig.php(316): OC\DB\QueryBuilder\QueryBuilder->execute()
#7 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppConfig.php(94): OC\AppConfig->loadConfigValues()
#8 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppConfig.php(272): OC\AppConfig->getApps()
#9 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(882): OC\AppConfig->getValues(false, 'installed_versi...')
#10 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(501): OC_App::getAppVersions()
#11 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(118): OC\Server->OC\{closure}(Object(OC\Server))
#12 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('OC\\Memcache\\Fac...')
#13 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppFramework/Utility/SimpleContainer.php(165): OC\ServerContainer->query('OC\\Memcache\\Fac...')
#14 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(114): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#15 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('MemCacheFactory')
#16 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(1541): OC\ServerContainer->query('MemCacheFactory')
#17 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(697): OC\Server->getMemCacheFactory()
#18 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(118): OC\Server->OC\{closure}(Object(OC\Server))
#19 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('OC\\App\\AppManag...')
#20 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/AppFramework/Utility/SimpleContainer.php(165): OC\ServerContainer->query('OC\\App\\AppManag...')
#21 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/3rdparty/pimple/pimple/src/Pimple/Container.php(114): OC\AppFramework\Utility\SimpleContainer->OC\AppFramework\Utility\{closure}(Object(OC\Server))
#22 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/ServerContainer.php(125): Pimple\Container->offsetGet('AppManager')
#23 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/Server.php(1731): OC\ServerContainer->query('AppManager')
#24 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(343): OC\Server->getAppManager()
#25 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/private/legacy/app.php(114): OC_App::getEnabledApps()
#26 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/base.php(640): OC_App::loadApps(Array)
#27 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/lib/base.php(1056): OC::init()
#28 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/console.php(46): require_once('/nix/store/i2xy...')
#29 /nix/store/i2xyg2ba0pkiwckm0q6w76skiz93m6vg-nextcloud-16.0.4/occ(11): require_once('/nix/store/i2xy...')


```
I am the single user of this nextcloud instance so it doesn't matter much. I wanted to share the error in case it was useful. Otherwise I wondered how I could fix it ? I thought about removing the nextcloud module and enabling it again so that it appears as a fresh install. But I am not sure it won't reuse the old state (aka "database").

cc @flokli @Ma27

**Comment by Ma27** [comment]
@teto in order to properly reproduce the issue, can you please tell us the nixpkgs rev you're on? :)

**Comment by teto** [comment]
it's a local checkout with some changes (but none to nextcloud).
`system: "x86_64-linux", multi-user?: yes, version: nix-env (Nix) 2.3pre20190712_aa82f8b, channels(teto): "", channels(root): "", nixpkgs: /home/teto/nixpkgs`

I've just tried running the nixops deploy `NIX_PATH=nixpkgs=channel:nixos-unstable ./dev-shell` (which would now be 54c76638733ca5e85aefea3caefd3286e8b537c5) with the same result.

**Comment by Ma27** [comment]
Ok, I just had a closer look at the isntaller's code and the NixOS module, to track down the actual issue I have the following questions:

* Which tables exist in your Nextcloud database? (`oc_appconfig` isn't something new, so I doubt that this is due to a broken migration currently)
* Which tableprefix is set? (if left empty or not declared, `oc_` is used by Nextcloud)
* Is it possible to give some more information about the stuff in `/var/lib/nextcloud/config/{,override.}config.php`? (Without secrets etc. of course)

**Comment by teto** [comment]
I suppose my db was in /var/lib/nextcloud/data/nextcloud.db (I have an owncloud.db next to it :s ) but when I load it with sqlite3 it looks empty "> .tables" returns nothing :/ is that the correct way to dump the schema.

Here are the config
```

cat /var/lib/nextcloud/config/config.php                                                                               ~
<?php
$CONFIG = array (
  'apps_paths' => 
  array (
    0 => 
    array (
      'path' => '/var/lib/nextcloud/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 => 
    array (
      'path' => '/var/lib/nextcloud/store-apps',
      'url' => '/store-apps',
      'writable' => true,
    ),
  ),
  'datadirectory' => '/var/lib/nextcloud/data',
  'skeletondirectory' => '',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'log_type' => 'syslog',
  'passwordsalt' => 'fake',
  'secret' => 'fake',
  'dbtype' => 'sqlite3',
  'version' => '16.0.1.1',
  'overwrite.cli.url' => 'http://localhost',
  'installed' => true,
  'instanceid' => 'oc8qd89r2bcd',
  'loglevel' => 2,
  'maintenance' => false,
  'log_level' => '0',
  'app_install_overwrite' => 
  array (
    0 => 'calendar',
  ),
  'trusted_domains' => 
  array (
    0 => 'nixos.fake.net',
  ),
);
```


```

cat /var/lib/nextcloud/config/override.config.php
<?php

$CONFIG = [
  'apps_paths' => [
    [ 'path' => '/var/lib/nextcloud/apps', 'url' => '/apps', 'writable' => false ],
    [ 'path' => '/var/lib/nextcloud/store-apps', 'url' => '/store-apps', 'writable' => true ],
  ],
  'datadirectory' => '/var/lib/nextcloud/data',
  'skeletondirectory' => '',
  'memcache.local' => '\OC\Memcache\APCu',
  'log_type' => 'syslog',
  'log_level' => '0',
  
  'dbname' => 'nextcloud',
  'dbhost' => 'localhost',
  
  'dbuser' => 'nextcloud',
  
  
  
  'dbtype' => 'sqlite',
  'trusted_domains' => ["nixos.fake.net"],
];


```

Is there a backup of the database somewhere ? (I can't access the online server anymore: Internal Server Error)

**Comment by Ma27** [comment]
That's indeed pretty weird :/

I'll set up a Nextcloud instance with sqlite3 locally during the day and report back.

**Comment by Ma27** [comment]
> Is there a backup of the database somewhere ? (I can't access the online server anymore: Internal Server Error)

AFAIK Nextcloud doesn't do automatic backups, I usually create regular, automated backups with some custom service.

Regarding your current problem: I tested a Nextcloud with `sqlite` in a VM and I didn't encounter any issues (including upgrade stuff). Do you recall the previous Nextcloud version you used (there's **maybe** something broken in their updater).

> I suppose my db was in /var/lib/nextcloud/data/nextcloud.db (I have an owncloud.db next to it :s )

Could you please explain to me why there's an `owncloud.db`? I just checked the Nextcloud code and the file is `<dbname>.db`, so there shouldn't(tm) be an owncloud database.

**Comment by teto** [comment]
Thanks for looking into it. 
I installed the nextcloud as a best effort only, I was not serious about the VM (used for experimentation, with nixos-unstable etc.). As I am going to lose access to the VM as well, I think I will just subscribe to fastmail and use their instance instead. I am not sure looking into the source will prove worthwile. Closing.

**Comment by Ma27** [comment]
Ack. In case anybody else stumbles upon this issue, feel free to ping me :)

**Comment by dermetfan** [comment]
I recently updated my nextcloud instance from NixOS 19.03 with state version 18.09 to NixOS 19.09 with state version 19.09 and faced the same issue. I noticed `owncloud.db` was several MB in size whereas `nextcloud.db` was just a few KB. After deleting `nextcloud.db` and renaming `owncloud.db` to `nextcloud.db` I was able to run `systemctl start nextcloud-setup` which successfully upgraded my database and fixed the installation. I guess somewhere in between these versions they renamed the database file.
